ks7/app/src/App.js

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react'
+import './App.css'
+import 'whatwg-fetch'
+
+import { AddTask, TodoList } from './Todo'
+
+class App extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      message: 'moon',
+      todoItems: []
+    }
+
+    this.onTaskAdded = this.onTaskAdded.bind(this)
+    this.onTaskDeleted = this.onTaskDeleted.bind(this)
+    this.onTaskUpdate = this.onTaskUpdate.bind(this)
+
+    this.fetchMessage()
+    this.fetchTodoItems()
+  }
+
+  fetchTodoItems(){
+    fetch('/api/todo/list', {
+      'Content': 'GET',
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    }).then(response => {
+      return response.json()
+    }).then(json => {
+      this.setState({ todoItems: json.todoList})
+    })
+  }
+
+  fetchMessage() {
+    fetch('/api/hello', {
+      'Content': 'GET',
+      headers: {
+        'Content-Type': 'application/json'
+      }
+    }).then(response => {
+      return response.json()
+    }).then(json => {
+      this.setState({ message: json.message })
+    })
+  }
+
+  onTaskAdded(taksName) {
+    const newItem = {
+      name: taksName,
+      done: false,
+      id: Date.now()
+    }
+
+    const payload = { newItem }
+
+    fetch('/api/todo/add', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(payload),
+    }).then(response => {
+      return response.json()
+    }).then(json => {
+      this.setState({ todoItems: json.todoList})
+    })
+  }
+
+  onTaskUpdate(item){
+    const itemToUpdate = this.state.todoItems.find(x => x.id === item.id)
+    itemToUpdate.done = !item.done
```

Comment:
Isn't this modifying the react `state` object? I think we need to do an `Object.assign()` call instead to make a copy of the object and update the copy.